### PR TITLE
Improve DST resilience

### PR DESCRIPTION
**Pull request by digitalcora**
**Asana:** [🚲 Make Trike DST resilient](https://app.asana.com/0/584764604969369/1201442435510236)

Since getting the message timestamp was the only reason we were doing any "parsing" of OCS messages, that is removed entirely. This does mean some invalid messages that were previously rejected will now be faithfully pushed to the stream — but I guess it is called "raw" for a reason, and in any case our parsing wouldn't have rejected _all_ invalid messages.

#### Reviewer Checklist
- [ ] Any new or changed functions have typespecs
- [ ] Tests were added for any new functionality (don't just rely on Lcov)
- [ ] Meets ticket's acceptance criteria

### CODE REVIEW

**Review by bklebe — APPROVED** [review]

**Comment by paulswartz** [comment]
If RTR is already parsing these ambiguous times, what is it doing in these cases? And can we replicate that behavior here so that every future client doesn't need to re-implement that logic?

**Comment by bklebe** [comment]
> If RTR is already parsing these ambiguous times, what is it doing in these cases? And can we replicate that behavior here so that every future client doesn't need to re-implement that logic?

@paulswartz it looks like currently it's just using a `NaiveDateTime` from Timex: https://github.com/mbta/rtr/blob/2987e003f05bfe2975e58013ff86602a3bcdb1aa/rtr/lib/ocs/parser.ex#L59

I spent some time messing around with different solutions while reviewing this, hoping I might be able to suggest something simple. I don't think there's a perfect solution. The ideal to me is converting these incoming local timestamps into valid UTC timestamps. One thought I had is like
```elixir
def disambiguate(first, second) do
  %{zone_abbr: zone} = DateTime.now!("America/New_York")

  zone
  |> disambiguate(first, second)
  |> DateTime.shift_zone!("Etc/UTC")
end

def disambiguate(zone, %{zone_abbr: zone} = first, _second), do: first
def disambiguate(zone, _first, %{zone_abbr: zone} = second), do: second
```
But this only works if we receive messages roughly when they're sent — we have no way of disambiguating them if they're really late.

**Comment by paulswartz** [comment]
Maybe worth checking with OPMI as well? Since they're processing the raw messages, maybe they're doing something here?

**Comment by paulswartz** [comment]
Here's a quick project I put together, which uses a zoned DateTime to support converting a NaiveDateTime: https://gitlab.com/paulswartz/safe_local_datetime

**Comment by losvedir** [comment]
> If RTR is already parsing these ambiguous times, what is it doing in these cases? 

In the case when the time is ambiguous, RTR just picks one of the times:

```elixir
  defp naive_to_datetime(naive_datetime, time_zone) do
    case DateTime.from_naive(naive_datetime, time_zone) do
      {:ok, dt} ->
        dt

      {:ambiguous, before_dt, _after_dt} ->
        Logger.warn("event=in_rtr_tz_ambiguous_dt ambiguous date time #{inspect(naive_datetime)}")
        before_dt

      {:gap, _before_dt, after_dt} ->
        Logger.warn("event=in_rtr_tz_gap gap datetime #{inspect(naive_datetime)}")
        after_dt
    end
  end
```

> Maybe worth checking with OPMI as well? Since they're processing the raw messages, maybe they're doing something here?

I believe they only care about revenue TMOVs, which we don't receive during the ambiguous time.

In any case, I'm in favor of doing the simplest possible thing here, and letting RTR parse times and such. I'd like to keep as much logic and complication off of opstech3 and that whole, crazy deployment situation as possible.

I'd like a future where we have an app closer to the OCC that does more parsing of the stream, and puts specific kinds of OCS messages into Kinesis, but down the road a bit when we have a better on-prem infrastructure story.

So, I'm in favor of the approach here.

I recently left [a comment on a realtime_signs PR](https://github.com/mbta/realtime_signs/pull/475#pullrequestreview-862889541) about switching to UTC, and it applies to Trike, too. Do you know if applying `utc_log: true` to the `Logger.Backend.Splunk` backend is enough to address that issue?

File: config/prod.exs [hunk 1]
```diff
@@ -3,7 +3,9 @@ import Config
 config :trike,
   kinesis_client: Trike.KinesisClient
 
-config :logger, backends: [Logger.Backend.Splunk, :console]
+config :logger,
+  backends: [Logger.Backend.Splunk, :console],
+  utc_log: true
```
[the remark this answers is not in the document]
Comment (digitalcora) [reply config/prod.exs:8]: Well, it was part of the acceptance criteria for the task, it wasn't on my own initiative 😄

**Comment by paulswartz** [comment]
> In any case, I'm in favor of doing the simplest possible thing here, and letting RTR parse times and such. I'd like to keep as much logic and complication off of opstech3 and that whole, crazy deployment situation as possible.

In that case, I'd suggest dropping the `received_time` entirely, rather than including the same value twice as duplicate data.

**Comment by digitalcora** [comment]
I also thought the `received_time` field now seems superfluous (per @bklebe comment). But since the format of these messages is specified in an accepted RFC and removing the field would technically (but not actually) break backwards compatibility, and the task didn't mention anything about that, I wasn't sure if that was the route we should go. Should we introduce a `com.mbta.ocs.raw_message.v2` event type, and update RTR to acknowledge it? Should we just sneakily remove the field without changing anything else, since RTR is the only consumer and it doesn't care about this field?

**Comment by paulswartz** [comment]
Normally I would say that it should be a new event type. However, as: 
1) RTR is the only consumer 
2) we didn't actually document the message format 
3) we aren't storing the historic data in this format

There isn't any need to provide backwards compatibility at this stage. If in the future we start storing these messages as CloudEvents somewhere, then yes, it would need to be a new message type.

**Comment by digitalcora** [comment]
> we didn't actually document the message format

[We sort of did](https://github.com/mbta/technology-docs/blob/main/rfcs/accepted/0004-socket-proxy-ocs-cloudevents.md#reference-level-explanation), though I guess technically the RFC was "standardize on CloudEvents" and details like the exact fields included in the `data` were non-normative. (maybe we should have done a better job delineating what was normative vs. not)

In any case, with @losvedir's blessing I will remove the field here and submit a quick amendment to the example in the RFC.

**Pull request merged**